### ISSUE TITLE
High idle CPU/scheduler usage due to :timer.apply_after calls

### ISSUE THREAD

**Issue by 1player** [IanLuites/data_daemon#4]
We've been noticing an idle 25% CPU scheduler usage in staging, when DataDaemon is active. 

Observer shows the process `timer_server` using most of the CPU time, which is the gen_server that powers the erlang :timer module. Killing this process restores a 0% idle CPU usage.

There's a call to `:timer.apply_after` in https://github.com/IanLuites/data_daemon/blob/72e0df0b12c244bbf5e9cff403ba33ce00050b25/lib/data_daemon/hound.ex#L217 and tracing this function call, shows it's called a ton of times, trace below:

```
(<0.22660.1888>) call timer:apply_after(600000,'Elixir.DataDaemon.Hound','host_check!',[<0.22660.1888>,"datadog_agent",8125,{10,0,7,27},ttl])
(<0.23963.1888>) call timer:apply_after(600000,'Elixir.DataDaemon.Hound','host_check!',[<0.23963.1888>,"datadog_agent",8125,{10,0,7,27},ttl])
(<0.23981.1888>) call timer:apply_after(600000,'Elixir.DataDaemon.Hound','host_check!',[<0.23981.1888>,"datadog_agent",8125,{10,0,7,27},ttl])
(<0.22771.1888>) call timer:apply_after(600000,'Elixir.DataDaemon.Hound','host_check!',[<0.22771.1888>,"datadog_agent",8125,{10,0,7,27},ttl])
(<0.23878.1888>) call timer:apply_after(600000,'Elixir.DataDaemon.Hound','host_check!',[<0.23878.1888>,"datadog_agent",8125,{10,0,7,27},ttl])
etc...
```

All of these are executed at the same time.

I seem to understand the logic between host_check!, but I'm not sure it's worth 25% of the CPU cycles, and it should probably default to disabled TTL refreshing, unless manually configured to do so.

Quoting from http://erlang.org/doc/efficiency_guide/commoncaveats.html

> Creating timers using erlang:send_after/3 and erlang:start_timer/3 , is much more efficient than using the timers provided by the timer module in STDLIB. The timer module uses a separate process to manage the timers. That process can easily become overloaded if many processes create and cancel timers frequently (especially when using the SMP emulator).

**Comment by 1player** [comment]
Confirmed that adding `config :app, Sample.DataDog, dns_refresh: :infinity` solves the idle CPU usage.

**Comment by IanLuites** [comment]
Hound DNS logic needed to be redone, so I've made it a bit sturdier.

I'll try to fix this too, because indeed it's not great.
I'm thinking of maybe moving it out of the `Hound` to limit the amount of resolves or doing it in the hound genserver.

**Comment by IanLuites** [comment]
Fixed in #6 

Most likely will be out today!
